app/serviceaccount.go

```diff
@@ -210,29 +226,61 @@ func NewServiceAccountCommand(getServiceAccountClientFn GetServiceAccountClientF
-							Required: true,
-							Aliases:  []string{"ar"},
+							Name:    accountRoleFlagName,
+							Usage:   fmt.Sprintf("The account role to set on the service account; valid types are: %v", accountActionGroups),
+							Aliases: []string{"ar"},
 						},
 						&cli.StringSliceFlag{
 							Name:    namespacePermissionFlagName,
 							Usage:   fmt.Sprintf("Flag can be used multiple times; value must be \"namespace=permission\"; valid types are: %v", namespaceActionGroups),
 							Aliases: []string{"np"},
 						},
+						&cli.StringFlag{
+							Name:    serviceAccountScopeType,
+							Usage:   fmt.Sprintf("The service account scope type; valid types are: %v", getScopeTypes()),
+							Aliases: []string{"st"},
+						},
+						&cli.StringFlag{
+							Name:    serviceAccountScopeID,
+							Usage:   "The entity ID that the service account is scoped to (e.g. namespace ID if creating a namespace scoped service account)",
+							Aliases: []string{"sid"},
+						},
 					},
 					Action: func(ctx *cli.Context) error {
 						if len(ctx.String(serviceAccountNameFlagName)) == 0 {
-							return fmt.Errorf("service account name must be provied with '--%s'", serviceAccountNameFlagName)
+							return fmt.Errorf("service account name must be provided with '--%s'", serviceAccountNameFlagName)
 						}
 
-						if len(ctx.String(accountRoleFlagName)) == 0 {
-							return fmt.Errorf("account role must be specified; valid types are %v", accountActionGroups)
+						scopeType := ctx.String(serviceAccountScopeType)
+						scopeID := ctx.String(serviceAccountScopeID)
+						if (len(scopeType) > 0) != (len(scopeID) > 0) {
+							return fmt.Errorf("both scope type and scope ID must be provided")
 						}
 
-						acctActionGroup, err := toAccountActionGroup(ctx.String(accountRoleFlagName))
-						if err != nil {
-							return fmt.Errorf("failed to parse account role: %w", err)
+						var scope *auth.ServiceAccountScope
+						if len(scopeType) > 0 {
+							t, ok := scopeTypes[scopeType]
+							if !ok {
+								return fmt.Errorf("invalid scope type: %s", scopeType)
+							}
+							scope = &auth.ServiceAccountScope{
+								Type: t,
+								Id:   scopeID,
+							}
+						}
+
+						var acctActionGroup auth.AccountActionGroup
+						if scope == nil {
+							if len(ctx.String(accountRoleFlagName)) == 0 {
+								return fmt.Errorf("account role must be specified; valid types are %v", accountActionGroups)
+							}
+							ag, err := toAccountActionGroup(ctx.String(accountRoleFlagName))
+							if err != nil {
+								return fmt.Errorf("failed to parse account role: %w", err)
+							}
+							acctActionGroup = ag
+						} else if scope.Type == auth.SERVICE_ACCOUNT_SCOPE_TYPE_NAMESPACE && len(ctx.String(accountRoleFlagName)) > 0 {
+							return fmt.Errorf("account role must be empty when creating a namespace scoped service account")
```

Comment:
nit: could be "namespace scoped service accounts may not have an account role" or something similar